scripts/o3de/o3de/android_support.py

```diff
@@ -0,0 +1,2098 @@
+def get_android_config(project_name: str or None) -> command_utils.O3DEConfig:
+    """
+    Create an android configuration a project. If a project name is provided, then attempt to look for the project and use its
+    project-specific settings (if any) as an overlay to the global settings. If the project name is None, then return an
+    android configuration object that only represents the global setting
+
+    :param project_name:    The name of the project to look for its project-specific setting. If None, only use the global settings.
+    :return: The android configuration object
+    """
+    return command_utils.O3DEConfig(project_name=project_name,
+                                    settings_filename=ANDROID_SETTINGS_FILE,
+                                    settings_section_name=ANDROID_SETTINGS_SECTION_NAME,
+                                    settings_description_list=SUPPORTED_ANDROID_SETTINGS)
+
+
+"""
+Map to make troubleshooting java issues related to mismatched java versions when running the sdkmanager.
+"""
+JAVA_CLASS_FILE_MAP = {
+    "65": "Java SE 21",
+    "64": "Java SE 20",
+    "63": "Java SE 19",
+    "62": "Java SE 18",
+    "61": "Java SE 17",
+    "60": "Java SE 16",
+    "59": "Java SE 15",
+    "58": "Java SE 14",
+    "57": "Java SE 13",
+    "56": "Java SE 12",
+    "55": "Java SE 11",
+    "54": "Java SE 10",
+    "53": "Java SE 9",
+    "52": "Java SE 8",
+    "51": "Java SE 7",
+    "50": "Java SE 6",
+    "49": "Java SE 5",
+    "48": "JDK 1.4",
+    "47": "JDK 1.3",
+    "46": "JDK 1.2"
+}
+
+ANDROID_HELP_REGISTER_ANDROID_SDK_MESSAGE = f"""
+Building android projects requires the android sdk manager command-line (https://developer.android.com/tools/sdkmanager). 
+Please follow the instructions at https://developer.android.com/studio to download the sdk manager command line tool from
+either Android Studio or the command line tools only and set the path to the 'sdkmanager{SDKMANAGER_EXTENSION}' path.
+
+If installing from Android Studio (recommended) then the default location of the Android SDK will be %HOME%   
+
+For example:
+
+o3de{O3DE_SCRIPT_EXTENSION} android-register --global --set-value android_sdk_root=<path to android SDK>
+
+"""
+
+class AndroidToolError(Exception):
+    pass
+
+# The ANDROID_GRADLE_PLUGIN_COMPATIBILITY_MAP manages the known android plugin known to O3DE and its compatibility requirements
+# Note: This map needs to be updated in conjunction with newer versions of the Android Gradle plugins.
+ANDROID_GRADLE_PLUGIN_COMPATIBILITY_MAP = [
+    ( ('8.1', ) , {'gradle': '8.0',
+                   'sdk_build_tools': '33.0.1',
+                   'jdk': '17',
+                   'url': 'https://developer.android.com/build/releases/gradle-plugin'} ),
+    ( ('8.0', ) , {'gradle': '8.0',
+                   'sdk_build_tools': '30.0.3',
+                   'jdk': '17',
+                   'url': 'https://developer.android.com/build/releases/past-releases/agp-8-0-0-release-notes'} )
+]
```

Comment:
This is probably better off as a map than a list of tuples, (tuples look unnecessary)